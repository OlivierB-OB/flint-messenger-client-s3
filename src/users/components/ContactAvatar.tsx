import { connect } from 'react-redux';
import { IAppState } from '../../appReducer';
import { UserAvatar } from './UserAvatar';

export interface IContactAvatarProps {
  target: string;
}

const mapStateToProps = ({ users, identity }: IAppState, { target }: IContactAvatarProps) => ({
  info: identity.info?._id === target ? identity.info : users.list.find((user) => user._id === target),
});

export const ContactAvatar = connect(mapStateToProps)(UserAvatar);
