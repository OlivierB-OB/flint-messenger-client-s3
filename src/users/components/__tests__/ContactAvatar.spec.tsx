import React from 'react';
import configureMockStore from 'redux-mock-store';
import thunk from 'redux-thunk';
import { shallow } from 'enzyme';
import { ContactAvatar } from '../ContactAvatar';

const middlewares = [thunk];
const mockStore = configureMockStore(middlewares);

describe('ContactAvatar', () => {
  let Connected: any = ContactAvatar;

  it('should show the contact avatar', async () => {
    const store = mockStore({
      identity: {},
      users: {
        list: [
          {
            _id: '123',
            firstName: 'foo',
            lastName: 'bar',
          },
        ],
      },
    });
    const component = shallow(<Connected store={store} target="123" />);
    expect(component).toMatchSnapshot();
  });

  it('should support unknown contacts', async () => {
    const store = mockStore({
      identity: {},
      users: {
        list: [
          {
            _id: '222',
            firstName: 'foo',
            lastName: 'bar',
          },
        ],
      },
    });
    const component = shallow(<Connected store={store} target="123" />);
    expect(component).toMatchSnapshot();
  });

  it('should works the profile as well', async () => {
    const store = mockStore({
      identity: {
        info: {
          _id: '123',
          firstName: 'foo',
          lastName: 'bar',
        },
      },
    });
    const component = shallow(<Connected store={store} target="123" />);
    expect(component).toMatchSnapshot();
  });
});
