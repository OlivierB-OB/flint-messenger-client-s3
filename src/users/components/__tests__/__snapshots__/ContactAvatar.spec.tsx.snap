// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ContactAvatar should show the contact avatar 1`] = `
<ContextProvider
  value={null}
>
  <UserAvatar
    dispatch={[Function]}
    info={
      Object {
        "_id": "123",
        "firstName": "foo",
        "lastName": "bar",
      }
    }
    store={
      Object {
        "clearActions": [Function],
        "dispatch": [Function],
        "getActions": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
      }
    }
    target="123"
  />
</ContextProvider>
`;

exports[`ContactAvatar should support unknown contacts 1`] = `
<ContextProvider
  value={null}
>
  <UserAvatar
    dispatch={[Function]}
    store={
      Object {
        "clearActions": [Function],
        "dispatch": [Function],
        "getActions": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
      }
    }
    target="123"
  />
</ContextProvider>
`;

exports[`ContactAvatar should works the profile as well 1`] = `
<ContextProvider
  value={null}
>
  <UserAvatar
    dispatch={[Function]}
    info={
      Object {
        "_id": "123",
        "firstName": "foo",
        "lastName": "bar",
      }
    }
    store={
      Object {
        "clearActions": [Function],
        "dispatch": [Function],
        "getActions": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
      }
    }
    target="123"
  />
</ContextProvider>
`;
